### DISCUSSION TITLE
Link component vs Imperative Navigation

### DISCUSSION THREAD

**Discussion by Beadle85** [expo/router#926] [Q&A]
Hey everyone, I'm a little confused about best practices when navigating inside my expo app using expo-router. Is there any reason to prefer wrapping a Button with <Link> versus passing router.push to the Button's onPress?

```
 <Link href="/other" asChild>
        <Button>Home</Button>
    </Link>
```

VS

`<Button onPress={() => router.navigate('/other')}`

Wrapping with Link seems more cumbersome but it also seems to be what the docs suggest doing. Imperative routing is the way to go if navigating outside of a React component. However, in a React component, is there a reason to choose one over the other?

**Comment by dmathisen** [comment]
Did you ever get any clarity here?

Edit: after some searching around, the only answers I can find ([here](https://www.reddit.com/r/Frontend/comments/v7pqm6/what_is_the_difference_between_routing_using/) and [here](https://www.reddit.com/r/reactjs/comments/ot5tvb/link_to_vs_onclick_history_push/)) is that using Link is a more accessible and semantically correct approach. If you're developing for mobile-only, it's likely less of an issue, but I'm not an accessibility guru, so who knows.
